### PR TITLE
Fix crash when running out of slots and discarding the submission.

### PR DESCRIPTION
**Pull request by florian-kuebler**
If we run out of timer slots, it may happen that we have not single
command buffer timer slot, and thus discard the submission (even
if that contained a valid "begin marker").

If the submission containing the "end marker" is not discarded,
we ran into a CHECK on the UI. This will discard the debug marker
on the UI.

Test: Capture Infiltrator for a long time since beginning.
Bug: http://177652631

### CODE REVIEW

File: src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp [hunk 1]
```diff
@@ -345,10 +345,13 @@ std::vector<TimerInfo> GpuQueueSubmissionProcessor::ProcessGpuDebugMarkers(
       } else {
         const GpuQueueSubmission* matching_begin_submission = FindMatchingGpuQueueSubmission(
             begin_marker_thread_id, begin_marker_post_submission_cpu_timestamp);
-        // Note that we receive submissions of a single queue in order (by CPU submission time). So
-        // if there is no matching "begin submission", the "begin" was submitted before the "end"
-        // and we lost the record of the "begin submission" (which should not happen).
-        CHECK(matching_begin_submission != nullptr);
+        // Note that we receive submissions of a single queue in order (by CPU submission time).
+        // However, if we are out of timer slot indices, we might discard submissions (if it
```
Comment (ronaldfw) [inline src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp:349]: Can we / should we discard these earlier?

**Review by ronaldfw — APPROVED** [review]
LGTM, modulo one question

File: src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp (hunk 1 above)
Comment (florian-kuebler) [reply src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp:349]: I don't know how. So I guess it is fine.

**Review by ronaldfw — APPROVED** [review]
LGTM, maybe add a brief additional comment (see suggestion)

File: src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp (hunk 1 above)
Comment (florian-kuebler) [reply src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp:349]: Exactly, this is the case.

File: src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp [hunk 2]
```diff
@@ -345,10 +345,13 @@ std::vector<TimerInfo> GpuQueueSubmissionProcessor::ProcessGpuDebugMarkers(
...
+        // contains no command buffer timers).
```
[the remark this answers is not in the document]
Comment (florian-kuebler) [reply src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp]: Slightly rephrased it, but in general makes sense.

[the remark this answers is not in the document]
Comment (ronaldfw) [reply src/OrbitCaptureClient/GpuQueueSubmissionProcessor.cpp]: Looks great, thanks!

**Pull request merged**